scripts/run_no_prep_tests.sh

```diff
@@ -13,9 +13,12 @@ flags()
         shift
     done
 }
+
 flags "$@"
 
 is_jdk_8=`echo $JAVA_HOME | grep 8.`
+is_jdk_14=`echo $JAVA_HOME | grep 14.`
```

Comment:
Thanks for the improvements! Some tests in slack-api-client project [fail](https://github.com/slackapi/java-slack-sdk/actions/runs/8430070548/job/23182070029) due to https://github.com/slackapi/java-slack-sdk/issues/892 (this is not a bug, just incompatibility with newer JDK versions, which affects only test code execution). Can you add is_jdk_17 and run tests only for `-pl bolt-micronaut`?

Reply:
At a quick glance, that issue may be similar to what I called out [here](https://github.com/slackapi/java-slack-sdk/pull/1294/files#diff-4e4b1306da55395df9a6891ba7aae7bd41fbd8ba1ec64e26507cdb76de2a5b77R174-R181). 

I'll go with checking `is_jdk_17` for now like you said, and maybe I can eventually take a look at that other issue as well.

Reply:
Actually, since `bolt-micronaut` depends on `slack-api-client` I'm not quite sure if that will work, as we will need to build that module in order to build `bolt-micronaut` if I am understanding correctly. I'm trying a few things locally right now, but my thought is that adding similar opening of jdk modules as I described above may be the simplest solution

Reply:
My understanding is that compiling all the modules never fails even with JDK 17 while some of them can fail when executing their test code. Thus, it should be feasible to compile slack-api-client (and others) and run only bolt-micronaut tests for now.

Reply:
Ive got the `run_no_prep_tests.sh` running locally on jdk 17 with very minor changes to the root `pom.xml`, I will quick push it up in case it is fine to just wrap in with these changes or else can try something else

Reply:
tried something new: added a new profile for when building with jdk17 to configure the surefire plugin to add the necessary `--add-opens`, tested on jdk 8 and jdk 17 locally and `./scripts/run_no_prep_tests.sh` ran fine on both

I tried with just changing command line arguments but I could not seem to quite figure out how to get maven to just compile the dependencies needed for bolt-micronaut and not run the tests for them in the process.

Reply:
@hrothwell Indeed, there is no simple way to do it, so I came up with a solution to do `mvn install` first and then run the tests with them: https://github.com/slackapi/java-slack-sdk/commit/ad0774e8bb91449447ff509083f9b2f52213beb1

Reply:
I thought the changes in the base `pom.xml` should have resolved the test issues as it worked for me on jdk 8 / 17 locally, I'm unsure exactly why the last few CodeCov tasks failed, but the tests all ran by the looks of it with the initial changes I had. So we could potentially mark #892 as complete also

Reply:
Oh I see. Will check if it works for me too later on!

Reply:
I did open #1296 which is the same changes I had originally so you wouldn't have to make the same changes again if you wanted to test it at all 👍